### ISSUE TITLE
5cb007b bug: Trainer pokemon edits are saved elsewhere

### ISSUE THREAD

**Issue by admiralkikan** [kwsch/pkNX#164]
On the forums I stated that some file data saved while trainer data did not. 

I decided to start with a clean folder and put all the saved data I had into the correct spots. Well, I edited Trainer 018 and it saved but I don't know where it got saved to. It didn't save to the folder for edited files. And it doesn't show up anywhere. But when I exit PKNX and load it back up, the edit is still present.

So what do you think is happening here? None of the files show a new modified date either. I can test the folder and see if that trainer shows up edited within the game. But I have my doubts about that for some reason.

**Comment by kwsch** [comment]
What file paths are you using for the program and the loaded data?

https://github.com/kwsch/pkNX/blob/5cb007b9e3986dca3b6e85cc0b1c175c0d001e32/pkNX.Containers/FileMitm.cs#L32-L50

Redirected read/writes is done ^; the only thing I could expect being funky is having weird paths that the string.Replace to swap locations fails.

**Comment by admiralkikan** [comment]
I grabbed the pre-built version on azure. Is that one a different commit than this one?

I put the Shield folder in the top of the pknx folder so the folder contains both the PKNX folder and game

Desktop\pkNX (2)\Pokemon Sword (0100ABF008968800)\0100ABF008968000\RomFS\bin <--- that is where the edited folders are and the edited files should be placed in those folders

The unedited folders for ExeFS and RomFS are in their normal places.

I can download this commit and put your new code in there and see if that fixes it. # @
![Screenshot 2021-08-24 162808](https://user-images.githubusercontent.com/24940637/130703161-44a92062-74d6-4c5f-946a-9491f37ff8c2.png)
![Screenshot 2021-08-24 162843](https://user-images.githubusercontent.com/24940637/130703172-0d5a9a7f-e91d-4775-84ed-40d3c073a29e.png)
![Screenshot 2021-08-24 163202](https://user-images.githubusercontent.com/24940637/130703173-df43d07e-8b70-4b34-8ad3-5a03669f2483.png)

**Comment by kwsch** [comment]
From your last screenshot, it looks like the trainer_poke is saving correctly to that folder. Only modified files are saved to the redirected folder, and the dump remains intact (unmodified).

**Comment by kwsch** [comment]
How are you launching pkNX? Double clicking the executable file?

**Comment by admiralkikan** [comment]
The last screenshot was from a month ago when I had to stop doing work on this. I only just picked it up again the other day and found none of the edits were being saved. The screenshot was to show that trainer 18 is missing in that folder yet, pknx is still pulling the edited data from somewhere.

Isn't trainer data just the data of the trainer themselves? IE the difficulty flags, items etc?

Yes, double clicking and comodo isn't tagging any of the files or pknx.

**Comment by kwsch** [comment]
Try:

1. Close pkNX
2. Move the "patch" content that pkNX has created to another folder.
3. Start pkNX, open your ROM
4. Make some changes, see if it makes a new folder next to the romfs

**Comment by admiralkikan** [comment]
Okay, I tested a few different things along with my old folders.

Everything removed: Editing trainer mon result in a new folder with the files showing up

So I then deleted that folder and put back my folder. For whatever reason, the bin files are saving again.

I think recycling the program with a new folder structure might fix the trainer_poke / Trainer_data save issue that everyone is having. Either that or this was a fluke. But I wont know until it happens again. If it does, I'll everything I just did and report back on it. But that could be in a few hours or a month from now.
